Felienne/hedy
# Enable dynamic keywords for teacher adventures

**Description**
In this PR we enable teacher to use dynamic keywords within their custom adventures. By supporting the use of curly brackets we can detect and parse keywords similar to how we approach this for all our other content. We verify that the placeholders are correct when trying to view a preview, making sure the teacher knows if they created a valid adventure.

**Fixes**
This PR fixes #3041

**How to test**
Verify that you can create, preview, view and customize adventures with keyword placeholders. Also verify that these are parsed correctly for the user selected keyword language. A teacher should **not** be able to save an adventure with an invalid placeholder and should receive a modal alert error instead, verify this as well.
